tests/cli/functional/conftest.py

```diff
@@ -120,6 +121,65 @@ def mock_click_confirm(mocker):
     return mocker.patch.object(click, 'confirm')
 
 
+@pytest.fixture(scope='function')
+def mock_input(mocker):
+    normal_input = MockInput()
+    hidden_input = MockInput()
+
+    mocker.patch('sys.stdin', new=normal_input)
+    mocker.patch('getpass.getpass', new=hidden_input)
+
+    return MockInputRouter(normal_input, hidden_input)
+
+
+class MockInputRouter:
```

Comment:
Move these two classes into `tests.mock`.

Reply:
Probably the fixture as well? I think they should be kept together

Reply:
Hmm - I think we'll have some fixture collection issues if we also move the fixtures there - I see your point though since its so local to this suite.

Reply:
If it wants to go there, and you can convince pytest to run the test, I think that's a good relocation aside from losing the "fixture" namespace.

At the moment, I've tried to deliberately separate pytest integrations from test doubles definitions.